pkg/skaffold/build/builder_mux.go

```diff
@@ -37,6 +37,11 @@ type BuilderMux struct {
 	byImageName map[string]PipelineBuilder
 	store       ArtifactStore
 	concurrency int
+	cache       Cache
+}
+
+type Cache interface {
+	AddArtifacts(ctx context.Context, bRes []graph.Artifact) error
```

Comment:
can this instead now be `AddArtifacts(ctx context.Context, bRes graph.Artifact) error` ?

Reply:
The function still accepts a slice of `graph.Artifact` and will add them all, so plural is appriorate:
https://github.com/GoogleContainerTools/skaffold/blob/c9d3521350f661cc32f587dad1c7ef39da21cc4d/pkg/skaffold/build/cache/retrieve.go#L174

Technically it is only being called in the singleton case, so I could refactor the function to make singular?

https://github.com/GoogleContainerTools/skaffold/blob/c9d3521350f661cc32f587dad1c7ef39da21cc4d/pkg/skaffold/build/builder_mux.go#L110-L114

Reply:
I think that we should refactor it to make it singular.

Reply:
Done.